parser_library/src/context/hlasm_context.cpp

```diff
@@ -153,37 +146,11 @@ auto time_sysvars(utils::timestamp now)
         std::string datc_val;
         std::string date_val;
         std::string systime;
-    } result;
-    auto& [datc_val, date_val, systime] = result;
-    auto year = std::to_string(now.year());
-    auto month = std::to_string(now.month());
-    auto day = std::to_string(now.day());
-    if (now.month() < 10)
-        month.insert(0, 1, '0');
-    if (now.day() < 10)
-        day.insert(0, 1, '0');
-
-    datc_val.reserve(8);
-    date_val.reserve(8);
-
-    datc_val.append(year);
-    datc_val.append(month);
-    datc_val.append(day);
-
-    date_val.append(month);
-    date_val.push_back('/');
-    date_val.append(day);
-    date_val.push_back('/');
-    date_val.append(year.c_str() + 2);
-
-    if (now.hour() < 10)
-        systime.push_back('0');
-    systime.append(std::to_string(now.hour()));
-    systime.push_back(':');
-    if (now.minute() < 10)
-        systime.push_back('0');
-    systime.append(std::to_string(now.minute()));
-
+    } result = {
+        std::format("{:04}{:02}{:02}", now.year(), now.month(), now.day()),
+        std::format("{:02}/{:02}/{:02}", now.month(), now.day(), now.year() % 100),
+        std::format("{:02}:{:02}:{:02}", now.hour(), now.minute(), now.second()),
```

Comment:
Just to be sure (and maybe I'm reading this wrong) - it seems like the `now.second()` wasn't used before and now it is. Is this intentional?

Reply:
good catch. It also is supposed to be using `.` and not `:` as the separator...